### PR TITLE
Ch 164592359 setting up development environment 2

### PR DESCRIPTION
**Pull request by Wabuluka**
#### What does this PR do?
Refactored code for the Epicmail from procedure code to OOP
#### Description of Task to be completed?
In this PR, I have provided for user signup, login endpoints and validators. I have also provided for create message , delete message, get all, get one, which are to be modified to any requirements. 
#### How should this be manually tested?
Use Postman with the following details
```
endpoint for creating a message:
http://127.0.0.1:5000/epicmail/api/v2/messages
```

```
{
    "data": {
        "address": "webbiewabuluka@gmail.com",
        "createdby": "dwabuluka@gmail.com",
        "createdon": "Fri, 22 Mar 2019 17:36:38 GMT",
        "id": 1,
        "message": "I had taken long with out seeing you so i decided to message you to see if we can schedule a meet up",
        "status": "sent",
        "subject": "Hello world"
    },
    "message": "Message has been created successfully",
    "status": 201
}
````
#### Screenshots (if appropriate)
message_1

@kenneth051 @eubule @Busobozihakim

### CODE REVIEW

File: epicmail/app/models/users.py [hunk 1]
```diff
@@ -5,132 +5,70 @@
...
-            }
-            return jsonify(error), 401
-        elif len(str(lastname)) < 5:
-            error = {
-                "status": 401,
-                "message":"Names accepted must be at least 5 characters"
+    def create_user(self,email, firstname, lastname, password):
+        if User.find_user_by_email(self.email):
+            return {
+                "status": 202,
```
Comment (kenneth051) [inline epicmail/app/models/users.py]: status code should be 409 please.409 means there is a conflict with the request a user is sending to the server

File: epicmail/app/views/message_views.py [hunk 2]
```diff
@@ -3,8 +3,9 @@
 from flask import Blueprint, request, make_response, jsonify
 from flask.views import MethodView
 from epicmail.app.views.user_views import LoginUser
-from epicmail.app.models.messages import Messages, messages
-# from epicmail.app.handler.auth import JwtAuth
+from epicmail.app.models.messages import Message
+from epicmail.app.handler.validators.message_validators import (
+    validate_address, validate_createdby, validate_message, validate_subject)
 
 # authentic = JwtAuth()
```
Comment (kenneth051) [inline epicmail/app/views/message_views.py]: remove the comments if there don't add any value to our code, please.

File: epicmail/app/views/message_views.py [hunk 3]
```diff
@@ -23,152 +24,57 @@ def post(self):
...
+            if validate_message(message):
+                return validate_message
+            if validate_createdby(createdby):
+                return validate_createdby
+            if validate_address(address):
+                return validate_address
+            
+            data = Message(subject, message, status, createdby, address)
+            msg = data.create_message(subject, message, status, createdby, address)
+            # print(msg)
```
Comment (kenneth051) [inline epicmail/app/views/message_views.py]: remove the comment here too please

File: epicmail/app/views/message_views.py [hunk 4]
```diff
@@ -23,152 +24,57 @@ def post(self):
...
+            return jsonify(msg), 201
 
 
 class GetSpecificMail(MethodView):
     """Get a message as specified by the user"""
     # @authentic.user_token
```
Comment (kenneth051) [inline epicmail/app/views/message_views.py]: remove the comment here too please

File: epicmail/app/models/users.py (hunk 1 above)
Comment (Wabuluka) [reply epicmail/app/models/users.py]: check

File: epicmail/app/views/message_views.py (hunk 2 above)
Comment (Wabuluka) [reply epicmail/app/views/message_views.py]: check

File: epicmail/app/views/message_views.py (hunk 3 above)
Comment (Wabuluka) [reply epicmail/app/views/message_views.py]: check

File: epicmail/app/views/message_views.py (hunk 4 above)
Comment (Wabuluka) [reply epicmail/app/views/message_views.py]: check

**Pull request merged**